packages-web/maps/src/components/GoogleMap.tsx

```diff
@@ -0,0 +1,162 @@
+import { createElement, ReactElement, useEffect, useRef } from "react";
+import classNames from "classnames";
+
+import googleApiWrapper from "./GoogleApi";
+import { getDimensions, getGoogleMapsStyles } from "../utils";
+import { Marker, SharedProps } from "../../typings";
+
+export interface GoogleMapsProps extends SharedProps {
+    scriptsLoaded?: boolean;
+    mapStyles?: string;
+
+    optionStreetView: boolean;
+    mapTypeControl: boolean;
+    fullScreenControl: boolean;
+    rotateControl: boolean;
+}
+
+export const GoogleMap = (props: GoogleMapsProps): ReactElement => {
+    const map = useRef<google.maps.Map>();
+    const googleMapsRef = useRef<HTMLDivElement>(null);
+    const defaultCenterLocation: google.maps.LatLngLiteral = { lat: 51.906688, lng: 4.48837 };
+    let bounds!: google.maps.LatLngBounds;
+
+    const markers = useRef<google.maps.Marker[]>([]); // Used to manage and remove markers from the map
+
+    useEffect(() => {
+        if (props.scriptsLoaded) {
+            createMap();
+        }
+    }, [props.scriptsLoaded]);
+
+    useEffect(() => {
+        addMarkers();
+    }, [props.locations, props.currentLocation]);
+
+    const createMap = (): void => {
+        const mapOptions: google.maps.MapOptions = {
+            zoom: props.zoomLevel,
+            zoomControl: props.optionZoomControl,
+            scrollwheel: props.optionScroll,
+            draggable: props.optionDrag,
+            streetViewControl: props.optionStreetView,
+            mapTypeControl: props.mapTypeControl,
+            fullscreenControl: props.fullScreenControl,
+            rotateControl: props.rotateControl,
+            minZoom: 1,
+            maxZoom: 20,
+            styles: getGoogleMapsStyles(props.mapStyles)
+        };
+        if (googleMapsRef.current && !map.current) {
+            map.current = new google.maps.Map(googleMapsRef.current, mapOptions);
+            bounds = new google.maps.LatLngBounds().extend(defaultCenterLocation);
```

Comment:
isn't this value always overridden? I guess it will become more clear when `addMarkers` is moved outside of `GoogleMap`